docs/resources/client/models/index.md

```diff
@@ -1,25 +1,295 @@
+
+Inside a `variants` block, there is an element for each blockstate. This is the predominant way of associating blockstates with models, used by the vast majority of blocks. The string representation of the blockstate (without the block name, so for example `"type=top,waterlogged=false"` for a non-waterlogged top slab, or `""` for a block with no properties) is the key, while the element is either a single model object or an array of model objects. If an array of model objects is used, a model will be randomly chosen from it. A model object consists of the following data:
+
+- `model`: A path to a model file location, relative to the namespace's `models` folder, for example `minecraft:block/cobblestone`.
+- `x` and `y`: Rotation of the model on the x-axis/y-axis. Limited to steps of 90 degrees. Optional each, defaults to 0.
+- `uvlock`: Whether to lock the UVs of the model when rotating or not. Optional, defaults to false.
+- `weight`: Only useful with arrays of model objects. Gives the object a weight, used when choosing a random model object. Optional, defaults to 1.
+
+In contrast, inside a `multipart` block, elements are combined depending on the properties of the blockstate. This method is mainly used by fences and walls, who enable the four directional parts based on boolean properties. A multipart element consists of two parts: a `when` block and an `apply` block.
+
+- The `when` block specifies either a string representation of a blockstate or a list of properties that must be met for the element to apply. The lists can either be named `"OR"` or `"AND"`, performing the respective logical operation on its contents. Both single blockstate and list values can additionally specify multiple actual values by separating them with `|` (for example `facing=east|facing=west`).
+- The `apply` block specifies the model object or an array of model objects to use. This works exactly like with a `variants` block.
+
+## Tinting
+
+Some blocks, such as grass or leaves, change their texture color based on their location and/or properties. [Model elements][elements] can specify a tint index on their faces, which will allow a color handler to handle the respective faces. The code side of things works through two events, one for block color handlers and one for item color handlers. They both work pretty similar, so let's have a look at a block handler first:
+
+```java
+@SubscribeEvent
+public static void registerBlockColorHandlers(RegisterColorHandlersEvent.Block event) {
+    // Parameters are the block's state, the level the block is in, the block's position, and the tint index.
+    event.register((state, level, pos, tintIndex) -> {
+            // Replace with your own calculation. See the BlockColors class for vanilla references.
+            // Generally, if the tint index is -1, it means that no tinting should take place
+            // and a default value should be used instead.
+            return 0xFFFFFF;
```

Comment:
Probably should be mentioned that vanilla implementations force the alpha to be 1f or 255.